src/ppx/grfx/vk/vk_descriptor.cpp

```diff
@@ -275,12 +276,25 @@ Result DescriptorSetLayout::CreateApiObjects(const grfx::DescriptorSetLayoutCrea
         vkBinding.stageFlags                   = ToVkShaderStageFlags(baseBinding.shaderVisiblity);
         vkBinding.pImmutableSamplers           = nullptr;
         vkBindings.push_back(vkBinding);
+
+        if (baseBinding.flags != 0) {
+            ValidateDescriptorBindingFlags(baseBinding.flags);
+            VkDescriptorBindingFlags vkBindingFlag = ToVkDescriptorBindingFlags(baseBinding.flags);
+            vkBindingFlags.push_back(vkBindingFlag);
+        }
```

Comment:
From the Vulkan spec:
[VkDescriptorSetLayoutCreateInfo](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkDescriptorSetLayoutCreateInfo.html)::pBindings[i] uses the flags in pBindingFlags[i]
I think it is possible for some flags to be 0 and for some flags to be not 0? So we have to insert 0 flags regardless in that case to preserve which flags apply to which binding

Reply:
oh right - my mistake. i'll add that change